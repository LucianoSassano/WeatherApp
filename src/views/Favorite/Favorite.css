#favorites-container {
  background: url(../../resources/pink-landscape.png) no-repeat;
  overflow: scroll;
  background-size: cover;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
  background-size: cover;
  background-position: center;
  min-height: 100vh;
  width: 100vw;
}

.favorites-data {
  text-align: center;
  height: 10vh;
  width: 100vw;
  align-content: center;
  color: #eae6ec;
  text-shadow: 1px 1px #212021;
  margin-bottom: 6%;
  background-color:RGBA(43,45,66,0.65);
}

.favorites-data h1 {
  font-size: 3vw;
}

.favorites-data h2 {
  font-size: 2vw;
}

.favorites-data h3 {
  font-size: 2vw;
}

@media all and (min-width: 280px) {
  #favorites-container .favorites-data {
  
    margin-bottom: 0.01vh;
    background-color:RGBA(43,45,66,0.65);
    height: 100%;
    margin-left: auto;
    margin-right: auto;
    width: 70vw;
  }

  #favorites-container .favorites-data h1 {
    font-size: 11vw;
  }
  #favorites-container .favorites-data h2 {
    font-size: 9vw;
  }
  #favorites-container .favorites-data h3 {
    font-size: 9vw;
  }

  #favorites-container .favorites-data .btn-danger {
    margin: 2%;
    font-size: 4vw;
  }
}

@media all and (min-width: 320px) {
  #favorites-container .favorites-data {
    
    margin-bottom: 0.01vh;
    background-color:RGBA(43,45,66,0.65);
    height: 100%;
    margin-left: auto;
    margin-right: auto;
    width: 70vw;
  }

  #favorites-container .favorites-data h1 {
    font-size: 10vw;
  }
  #favorites-container .favorites-data h2 {
    font-size: 9vw;
  }
  #favorites-container .favorites-data h3 {
    font-size: 9vw;
  }
}

@media all and (min-width: 375px) {
  #favorites-container .favorites-data {
    margin-top: 11vh;
    margin-bottom: 1vh;
    background-color:RGBA(43,45,66,0.65);
    height: 100%;
    margin-left: auto;
    margin-right: auto;
    width: 70vw;
  }

  #favorites-container .favorites-data h1 {
    font-size: 8vw;
  }
  #favorites-container .favorites-data h2 {
    font-size: 6vw;
  }
  #favorites-container .favorites-data h3 {
    font-size: 6vw;
  }
}

@media all and (min-width: 425px) {
  #favorites-container .favorites-data {
   
    margin-bottom: 0.3vh;
   
    background-color:RGBA(43,45,66,0.65);
    height: 100%;
    margin-left: auto;
    margin-right: auto;
    width: 70vw;
  }

  #favorites-container .favorites-data h1 {
    font-size: 8vw;
  }
  #favorites-container .favorites-data h2 {
    font-size: 6vw;
  }
  #favorites-container .favorites-data h3 {
    font-size: 6vw;
  }
}

@media all and (min-width: 768px) {


  #favorites-container .favorites-data {
    margin-top: 2.5vh;
    margin-bottom: 0.3vh;
    background-color:RGBA(43,45,66,0.65);
    height: 100%;
    margin-left: auto;
    margin-right: auto;
    width: 40vw;
  }

  #favorites-container .favorites-data h1 {
    font-size: 8vw;
  }
  #favorites-container .favorites-data h2 {
    font-size: 6vw;
  }
  #favorites-container .favorites-data h3 {
    font-size: 6vw;
  }

  #favorites-container .favorites-data .btn-danger {
    margin: 2%;
    font-size: 3vw;
  }
}

@media all and (min-width: 1024px) {
  #favorites-container .favorites-data {
  
  
    margin-bottom: 0.3vh;
    background-color:RGBA(43,45,66,0.65);
    height: 100%;
    margin-left: auto;
    margin-right: auto;
    width: 30vw;

  }

  #favorites-container .favorites-data h1 {
    font-size: 6vw;
  }
  #favorites-container .favorites-data h2 {
    font-size: 4vw;
  }
  #favorites-container .favorites-data h3 {
    font-size: 4vw;
  }

  #favorites-container .favorites-data .btn-danger {
    margin: 2%;
    font-size: 2vw;
  }
}

@media all and (min-width: 1440px) {
  #favorites-container .favorites-data {


    margin-bottom: 3vh;
    background-color:RGBA(43,45,66,0.65);
    height: 60vh;
    margin-left: auto;
    margin-right: auto;
    width: 30vw;

  }

  #favorites-container .favorites-data h1 {
    font-size: 6vw;
  }
  #favorites-container .favorites-data h2 {
    font-size: 4vw;
  }
  #favorites-container .favorites-data h3 {
    font-size: 4vw;
  }

  #favorites-container .favorites-data .btn-primary {
    margin: 4%;
    font-size: 3vw;
  }
}

@media all and (min-width: 1920px) {
  #favorites-container .favorites-data {
    margin-bottom: 0.03vh;
    background-color:RGBA(43,45,66,0.65);
    height: 50vh;
    margin-left: auto;
    margin-right: auto;
    width: 25vw;

  }

  #favorites-container .favorites-data h1 {
    font-size: 4vw;
  }
  #favorites-container .favorites-data h2 {
    font-size: 3vw;
  }
  #favorites-container .favorites-data h3 {
    font-size: 3vw;
  }

  #favorites-container .favorites-data .btn-danger {
    margin: 1%;
    font-size: 1vw;
  }
}
