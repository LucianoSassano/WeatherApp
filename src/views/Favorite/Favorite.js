import React, { useState, useEffect } from "react";
import "./Favorite.css";

const Favorite = () => {
  const [favs, setFavs] = useState("");

  return (
    <>
      <div className="Favorites-container">
        <h1>FAVS VIEW</h1>
      </div>
    </>
  );
};

export default Favorite;
