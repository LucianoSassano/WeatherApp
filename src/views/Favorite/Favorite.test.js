import React from 'react';
import { render, cleanup, screen} from '@testing-library/react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import Favorite from './Favorite.js';


afterEach(cleanup);


it('Favorite renders correctly', () => {

    const div = document.createElement("div");
    ReactDOM.render(<Favorite></Favorite>, div)
})


it('Favorite delete button renders correctly', async () => {

    render(<div data-testid="favoriteWeather" className="weather-container">

        <div className="container-data" aria-label="weather-result">
            <h1
                aria-label="city name and country of city"
                value={(location.city, location.country)}
            >
                {location.city},{location.country}
            </h1>
            <h3
                aria-label="location temperature in celcius degrees"
                value={location.temp}
            >
                {location.temp + "Â°C"}
            </h3>
            <h2
                aria-label="location weather description"
                value={location.description}
            >
                {location.description}
            </h2>
            <button

                data-testid ="deleteBtn"
                type="button"
                className="btn btn-danger"
               
                aria-label="delete forecaste from favorites"
            >
                Delete
        </button>
        </div>
  
  </div>)

    

    expect(screen.getByTestId('deleteBtn')).toHaveTextContent("Delete")   
    expect(screen.getByTestId('deleteBtn')).toBeInTheDocument() 


})



it("Favorite component matches snapshot", () => {

    const favorite = renderer.create(<Favorite></Favorite>).toJSON();
    expect(favorite).toMatchSnapshot();

})

