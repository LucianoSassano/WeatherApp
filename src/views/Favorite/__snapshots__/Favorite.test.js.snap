// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite component matches snapshot 1`] = `
<div
  className="favorites-container"
>
  <div
    className="navbar-container"
  >
    <nav
      aria-label="navbar"
      className="navbar fixed-top navbar-expand-lg navbar-light bg-light"
      data-testid="navbarTestId"
    >
      <div
        className="container-fluid"
        id="navbarSupportedContent"
      >
        <div>
          <ul
            className="navbar-nav mr-auto"
          >
            <li
              aria-label="Home button"
              className="nav-item "
            >
              <a
                className="nav-link"
                href="./"
              >
                Home
              </a>
            </li>
            <li
              aria-label="Favorites button"
              className="nav-item"
            >
              <a
                className="nav-link "
                href="./favorites"
              >
                Favorites
              </a>
            </li>
          </ul>
        </div>
        <div
          className="searchForm"
        >
          <div
            className="custom-control custom-switch custom-switch-xl"
          >
            <input
              aria-label="dual location search toggle button"
              className="custom-control-input"
              id="customSwitch1"
              onClick={[Function]}
              type="checkbox"
            />
            <label
              className="custom-control-label"
              data-testid="dualSearchBtn"
              htmlFor="customSwitch1"
            >
              Dual Location Search
            </label>
          </div>
          <div
            className="single-search-container"
          >
            <form
              className="form-inline my-2 my-lg-0"
            >
              <input
                aria-label="Search a location"
                className="form-control mr-sm-2"
                placeholder="Search a location"
                type="search"
              />
              <button
                aria-label="search"
                className="btn btn-outline-success my-2 my-sm-0"
                type="button"
              >
                Search
              </button>
            </form>
          </div>
        </div>
      </div>
    </nav>
  </div>
  <div>
    <div
      aria-label="forecast list "
      className="container"
    >
      <div />
    </div>
  </div>
</div>
`;
